prediction_market_agent/db/sql_handler.py

```diff
@@ -11,31 +14,41 @@
 
 class SQLHandler:
     def __init__(
-        self, model: t.Type[SQLModelType], sqlalchemy_db_url: str | None = None
+        self,
+        model: t.Type[SQLModelType],
+        sqlalchemy_db_url: str | None = None,
     ):
-        self.engine = create_engine(
+        self.sqlalchemy_db_url = (
             sqlalchemy_db_url
-            if sqlalchemy_db_url
-            else check_not_none(DBKeys().SQLALCHEMY_DB_URL)
+            if sqlalchemy_db_url is not None
+            else check_not_none(DBKeys().SQLALCHEMY_DB_URL).get_secret_value()
         )
+
+        api_keys = APIKeys_PMAT(SQLALCHEMY_DB_URL=SecretStr(self.sqlalchemy_db_url))
```

Comment:
Kudos for switching to DBManager! But what about just doing

```python
class SQLHandler:
    def __init__(
        self,
        model: t.Type[SQLModelType],
        api_keys: APIKeys | None = None,
    ):
        self.db_manager = DBManager(api_keys)
```

instead of all the if else, converting to normal string and then back to a secret, etc.?

Reply:
The casting magic I have to do is this
```
    def __init__(
        self,
        model: t.Type[SQLModelType],
        sqlalchemy_db_url: str | None = None,
    ):
        self.sqlalchemy_db_url = (
            sqlalchemy_db_url
            if sqlalchemy_db_url is not None
            else check_not_none(DBKeys().SQLALCHEMY_DB_URL).get_secret_value()
        )

        api_keys = APIKeys_PMAT(SQLALCHEMY_DB_URL=SecretStr(self.sqlalchemy_db_url))
        self.db_manager = DBManager(api_keys)
```
Observe that APIKeys_PMAT needs to be provided. I guess it's just a matter of doing this logic once (here, as is) or everywhere else where the `__init__` is called. I still think this approach (not very elegant but working logic) is the one to be used here.

Reply:
Then why not go one level deeper and change `DBManager` to accept `sqlalchemy_db_url` instead of `APIKeys`?

Then it can be just 

```
def __init__(
        self,
        model: t.Type[SQLModelType],
        sqlalchemy_db_url: str | None = None,
    ):
        self.db_manager = DBManager(sqlalchemy_db_url)
```

I mean, yes this obviously works, but c'mon 😄

Reply:
Sure, see PR https://github.com/gnosis/prediction-market-agent-tooling/pull/571